layer/executables/packager.py

```diff
@@ -0,0 +1,154 @@
+import inspect
+import os
+import pickle  # nosec
+import shutil  # nosec
+import sys
+import tempfile
+import zipapp
+from pathlib import Path
+from typing import Any, Callable, List, Optional
+from uuid import uuid4
+
+import cloudpickle  # type: ignore
+
+
+def package_function(
+    function: Callable[..., Any],
+    resources: Optional[List[Path]] = None,
+    pip_dependencies: Optional[List[str]] = None,
+    output_dir: Optional[Path] = None,
+) -> Path:
+    """Packages layer function as a Python executable."""
+
+    with tempfile.TemporaryDirectory() as source_dir:
+        source = Path(source_dir)
+
+        requirements_path = source / "requirements.txt"
+        with open(requirements_path, mode="w", encoding="utf8") as requirements:
+            requirements.write(
+                "\n".join(_runtime_requirements + (pip_dependencies or []))
+            )
+
+        main_path = source / "__main__.py"
+        with open(main_path, mode="w", encoding="utf8") as main:
+            main.write(_loader_source())
+
+        if resources:
+            _package_resources(source, resources)
+
+        function_path = source / "function"
```

Comment:
nit: I'd put `"function"` into a constant above.

Reply:
Not worth it, would require to redefine it in `_loader_source` anyway.